scripts/generate_changelog.rb

```diff
@@ -1,100 +1,146 @@
-# Any notes that do not have a valid label are ignored and not added
-# to the changelog.
-entries = {
-  'feature'     => { name: 'New features', entries: [] },
-  'bug'         => { name: 'Bug fixes',    entries: [] },
-  'deprecation' => { name: 'Deprecations', entries: [] },
-  'removal'     => { name: 'Removals',     entries: [] }
-}
-
-version    = ARGV.first
-changelog  = File.expand_path('CHANGELOG.md', Dir.pwd)
-
-unless version
-  warn "Usage: generate_changelog.rb VERSION"
-  exit 1
-end
+class ChangelogGenerator
+  attr_reader :version, :entries
 
-unless File.file?(changelog)
-  warn "Could not find changelog at #{changelog}"
-  exit 1
-end
+  def initialize(version)
+    unless version
+      warn "Usage: generate_changelog.rb VERSION"
+      exit 1
+    end
+
+    @version = version
+    @entries = {
+      'feature'     => { name: 'New features', entries: [] },
+      'bug'         => { name: 'Bug fixes',    entries: [] },
+      'deprecation' => { name: 'Deprecations', entries: [] },
+      'removal'     => { name: 'Removals',     entries: [] }
+    }
+
+    # Setting the changelog path early lets us check that it exists
+    # before we spend time making API calls
+    changelog
+  end
 
-# Get the latest tag
-latest = `git describe --abbrev=0`.chomp
-
-# Read the git log between master and the most recent tagged release
-# The log is formatted to only include the body of a commit and are
-# terminated with a null terminator. These entries are then filtered
-# to only include those that have a valid LABEL (e.g. !feature, !bug)
-stdout_str, stderr_str, status = Open3.capture3(
-  "git log -z HEAD...#{latest} "\
-  "--pretty=format:'%B' "\
-  "--grep='!(#{entries.keys.join('|')})' -E"
-)
-
-if status.success?
-  # Create an array of commit messages
-  commits = stdout_str.split("\u0000")
-
-  if commits.empty?
-    warn "Did not find any changelog entries"
-    exit 0
+  def labels
+    @entries.keys
   end
 
-  # Parse the notes and map them to the correct list of entries
-  commits.each do |commit|
-    label, entry = parse_commit(entries.keys, commit)
+  def client
+    unless @client
+      unless ENV['GITHUB_TOKEN']
+        warn "Missing GitHub personal access token. Set $GITHUB_TOKEN with a "\
+             "personal access token to use this script."
+        exit 1
+      end
+
+      Octokit.configure do |c|
+        c.auto_paginate = true
+      end
 
-    if entries.key?(label)
-      entries[label][:entries] << entry
+      @client = Octokit::Client.new(access_token: ENV['GITHUB_TOKEN'])
     end
+
+    @client
   end
-else
-  warn stderr_str
-  exit 1
-end
 
-# Build the new changelog content from the list of valid notes
-new_content = <<~CONTENT
-  # Changelog
+  def latest
+    @latest ||= client.tags('puppetlabs/bolt').first.name
+  end
 
-  ## Bolt #{version} (#{Time.now.strftime '%Y-%m-%d'})
+  def commits
+    @commits ||= client.compare('puppetlabs/bolt', latest, 'master').commits
```

Comment:
Since we don't currently have any external contributions merged in post-2.6.0, you can change `latest` to an earlier Bolt version like `'2.3.0'` to see them.

Reply:
I know it'd be a bit of manual work but can you include this update since 2.0 in the CHANGELOG in this PR?

Reply:
Sure, it's simple enough. :)